### PR TITLE
Tunnel map items as objects proposal

### PR DESCRIPTION
**Pull request by kcudnik**
Purpose for this change is to remove bridge id from sai_tunnel_map_params_t and replace it with new object type to keep same feature. Current solution will require very special logic to handle bridge_if id in sai_tunnel_map_params_t in metadata in evey place where it will be used, and its not uniform with all other objects in SAI

This change could be much simpler if there would be requirement that when bridge id is used, then same bridge must be used for all tunnel map, then this bridge_if id could be moved tunnel itself instead of tunnel map

### CODE REVIEW

**Comment by kcudnik** [comment]
This is reissued commit since preveious was merged by mistake by @lguohan

File: inc/saitunnel.h [hunk 1]
```diff
@@ -61,6 +61,137 @@ typedef enum _sai_tunnel_map_type_t
...
+     */
+    SAI_TUNNEL_MAP_ITEM_ATTR_END,
+
+    /** Custom range base value */
+    SAI_TUNNEL_MAP_ITEM_ATTR_CUSTOM_RANGE_START = 0x10000000,
+
+    /** End of custom range base */
+    SAI_TUNNEL_MAP_ITEM_ATTR_CUSTOM_RANGE_END
+
+} sai_tunnel_map_item_attr_t;
```
Comment (tushar-ty) [inline inc/saitunnel.h]: Could you call it map_'entry' or something else instead of 'item' ?

Comment (kcudnik) [inline inc/saitunnel.h]: yes, i will update that

Comment (tushar-ty) [inline inc/saitunnel.h]: Thanks. 
Just FYI, my input was for the entire file not just this instance.

**Comment by MELLANOX-zhichenr** [comment]
Is there a reason to keep the old tunnel map object? Can it be fully replaced by the new tunnel map entry object?

**Comment by kcudnik** [comment]
Idea was to not cause too much trouble for current vendor implementations. We can remove that in future eventually.

**Comment by MELLANOX-zhichenr** [comment]
I see. Thanks!

**Comment by MELLANOX-zhichenr** [comment]
Hi I have several questions:

1. Tunnel has an attribute called encap tunnel map. Could this attribute be either 'tunnel map' type or 'tunnel map entry' type? If yes then the metadata might need to be updated. If no, then how to associate tunnel map entry with tunnel?

2. Tunnel map entry has an attribute called tunnel map. Does this attribute need to be mandatory? If yes, then does setting oecn value on tunnel map entry changes the attribute in tunnel map?

3. In tunnel map, comment on tunnel map list said 'if empty, tunnel map can be used to assign tunnel map entry objects'.  Is there an example of how to do the assignment?

4. For SAI_TUNNEL_MAP_ENTRY_ATTR_OECN_KEY, should the condition be only SAI_TUNNEL_MAP_ENTRY_ATTR_TUNNEL_MAP_TYPE == SAI_TUNNEL_MAP_TYPE_OECN_TO_UECN since it is always 'key to value'?

**Pull request merged**